v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/transforms/KafkaTransform.java

```diff
@@ -97,10 +95,8 @@ public static PTransform<PBegin, PCollection<KV<byte[], GenericRecord>>> readAvr
             .withKeyDeserializerAndCoder(
                 ByteArrayDeserializer.class, NullableCoder.of(ByteArrayCoder.of()))
             .withValueDeserializer(new KafkaSchemaDeserializerProvider(avroSchema))
-            .withConsumerConfigUpdates(config);
-    if (sslConfig != null) {
-      kafkaRecords = kafkaRecords.withConsumerFactoryFn(new FileAwareConsumerFactoryFn());
-    }
+            .withConsumerConfigUpdates(config)
+            .withConsumerFactoryFn(new FileAwareConsumerFactoryFn());
```

Comment:
Would you mind fixing the other inconsistency in `readAvroFromKafka` as well? That method does not apply the effect of setting `--enableCommitOffsets`, see the suggestion below.

A clean up nit: The `enableCommitOffsets` parameter is typed as a boxed primitive (`Boolean`), but none of the method bodies for `KafkaTransform.read*` require any property of that type (e.g. nullness) so the parameter type can be replaced with its primitive counterpart (`boolean`).

```
  /**
   * Configures Kafka consumer that reads Avro GenericRecord.
   *
   * @param bootstrapServers Kafka servers to read from
   * @param topicsList Kafka topics to read from
   * @param config configuration for the Kafka consumer
   * @param enableCommitOffsets applies {@link KafkaIO.Read#commitOffsetsInFinalize()} if set
   * @param avroSchema schema for the Kafka record's value
   * @return PCollection of Kafka Key & Value Pair deserialized in string format
   */
  public static PTransform<PBegin, PCollection<KV<byte[], GenericRecord>>> readAvroFromKafka(
      String bootstrapServers,
      List<String> topicsList,
      Map<String, Object> config,
      Boolean enableCommitOffsets,
      String avroSchema) {
    KafkaIO.Read<byte[], GenericRecord> kafkaRecords =
        KafkaIO.<byte[], GenericRecord>read()
            .withBootstrapServers(bootstrapServers)
            .withTopics(topicsList)
            .withKeyDeserializerAndCoder(
                ByteArrayDeserializer.class, NullableCoder.of(ByteArrayCoder.of()))
            .withValueDeserializer(new KafkaSchemaDeserializerProvider(avroSchema))
            .withConsumerConfigUpdates(config)
            .withConsumerFactoryFn(new FileAwareConsumerFactoryFn());

    if (enableCommitOffsets) {
      kafkaRecords = kafkaRecords.commitOffsetsInFinalize();
    }
    return kafkaRecords.withoutMetadata();
  }
```

Reply:
the method `readAvroFromKafka` isn't used in the `KafkaToBigQueryFlex` template, hence I didn't make any changes to the method.

Sent the cleanup on a different [pr](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1901)

Reply:
commitOffsetsInFinalize changes have been added to this PR.